brokers/artifacts/extensions.go

```diff
@@ -31,20 +31,24 @@ func (b *Broker) ProcessPlugin(plugin *model.CachedPlugin) error {
 		return err
 	}
 
-	var infoLine strings.Builder
-	defer b.PrintInfoBuffer(&infoLine)
+	// Workaround: messages can be displayed out of order in the workspace loading page
+	// Collect messages in a buffer and print them as a single string when needed.
+	logBuf := make([]string, 0)
+	defer b.flushLog(logBuf)
 
-	infoLine.WriteString(fmt.Sprintf("Processing plugin %s", plugin.ID))
+	logBuf = append(logBuf, fmt.Sprintf("Processing plugin %s", plugin.ID))
 	numExtensions := len(plugin.CachedExtensions)
 	extensionIdx := 0
 	for URL, path := range plugin.CachedExtensions {
 		extensionIdx = extensionIdx + 1
-		infoLine.WriteString(fmt.Sprintf("\n  Installing plugin extension %d/%d", extensionIdx, numExtensions))
+		logBuf = append(logBuf, fmt.Sprintf("  Installing plugin extension %d/%d", extensionIdx, numExtensions))
 		if path != "" {
-			infoLine.WriteString(fmt.Sprintf("\n    Plugin already downloaded"))
+			logBuf = append(logBuf, fmt.Sprintf("    Plugin already downloaded"))
+			b.flushLog(logBuf)
```

Comment:
When plugin is already downloaded iteration should be really fast, I guess we don't really need to flush buffer here.
But I'm OK with flushing it here if it does not lead to propagating logs in the wrong order.

Reply:
You're right, it was actually a mistake in `b.flushLog` that caused me to put this line here (the deferred flush gets the just-initialized buffer so it always logged nothing). This is fixed now.